OMAR - Cesar/Vigenere Encryption [Final Kata] [FINISHED]

omar-limbert opened the pull request:
- Encode with Cesar Encryption Ready.
- Decode with Cesar Encryption Ready.
- Encode wiith Vigenere Encryption Ready.
- Decode with Vigenere Encryption Ready.
- Supports Null message, Empty message, Negative Key, Special Characters, Blank Spaces.

Commit:
CESAR ENCRYPTION [FINAL KATA]

- Encode Ready.
- Decode Ready.
- Improve code and Vigenere Encryption coming soon.

Commit:
TESTS ADDED

- Encode with Cesar Encryption Ready.
- Decode with Cesar Encryption Ready.
- Enconde and Decode with Vigenere Encryption in progress.

carledriss commented on src/main/java/org/fundacionjala/coding/omar/CesarEncryption.java:
```diff
@@ -0,0 +1,89 @@
+    private int getAsciiIndex(int ascii) {
+        return ascii > ASCII_MAX
+                ? ascii - DIFFERENCE
+                : ascii < ASCII_MIN ? ascii + DIFFERENCE
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested ternary operation into an independent statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3358)

carledriss commented:
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

Commit:
sonar fix

sonar fix

Commit:
sonar fix

Commit:
FINAL KATA COMPLETED

- Encode with Cesar Encryption Ready.
- Decode with Cesar Encryption Ready.
- Encode with Vigenere Encryption Ready.
- Decode with Vigenere Encryption Ready.
- Support Null, Blank Spaces, Negative Key, Special Characters.

carledriss commented on src/test/java/org/fundacionjala/coding/omar/CesarEncryptionTest.java:
```diff
@@ -0,0 +1,201 @@
+     */
+    @Test
+    public void testEncodeVigenereEncryption() {
+        assertEquals("VJYPGAOG", enconde.encodeMessageWithVigenereEncryption("JALASOFT", "LIMON"));
```
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "LIMON" 8 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)

carledriss commented on src/main/java/org/fundacionjala/coding/omar/CesarEncryption.java:
```diff
@@ -0,0 +1,178 @@
+        return message == null || message.trim().isEmpty()
+                ? ""
+                : Arrays.stream(message.split(""))
+                .map(letter -> this.isValidCharacter(letter)
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested ternary operation into an independent statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3358)

carledriss commented on src/main/java/org/fundacionjala/coding/omar/CesarEncryption.java:
```diff
@@ -0,0 +1,178 @@
+        return message == null || message.trim().isEmpty()
+                ? ""
+                : Arrays.stream(message.split(""))
+                .map(letter -> this.isValidCharacter(letter)
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested ternary operation into an independent statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3358)

carledriss commented:
SonarQube analysis reported 3 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 2 major

Watch the comments in this conversation to review them.

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

Commit:
fix

- Encode with Cesar Encryption Ready.
- Decode with Cesar Encryption Ready.
- Encode wiith Vigenere Encryption Ready.
- Decode with Vigenere Encryption Ready.
- Supports Null message, Empy message, Negative Key, Special Characters, Blank Spaces.

carledriss commented on src/main/java/org/fundacionjala/coding/omar/CesarEncryption.java:
```diff
@@ -0,0 +1,193 @@
+ * @author Omar Limbert Huanca Sanchez - AT-[06].
+ * @version 1.0.
+ */
+public class CesarEncryption {
```
@AT-06/at-06 
CesarEncryption ?
The class is doing two kind of encryption.
Split in two classes and apply abstraction with encode and decode abstract methods.

Commit:
Improved Code

Improved Code

Commit:
fix

fix

Commit:
fix

fix

Commit:
abstract methods added

abstract methods added

Commit:
code coverage incremented

code coverage incremented

carledriss commented on src/main/java/org/fundacionjala/coding/omar/encryption/VigenereEncryption.java:
```diff
@@ -0,0 +1,101 @@
+     */
+    public VigenereEncryption() {
+        super();
+        indexVigenereKey = 0;
```
this line of code is redundant since integer attributes of class are by default zero

carledriss commented on src/test/java/org/fundacionjala/coding/omar/encryption/EncryptionTest.java:
```diff
@@ -0,0 +1,206 @@
+    public void init() {
+
+        cesarEncryption = new CesarEncryption();
+        vigenereEncryption = new VigenereEncryption();
```
This is wrong
because you are not using the two instances in all the tests

Consider splitting in two unit test class

Commit:
Final Fix

Final Fix
